### DISCUSSION TITLE
Can I use cylindrical coordinate in adjoint solver as well?

### DISCUSSION THREAD

**Discussion by semiuse** [NanoComp/meep#2537] [Q&A]
Hello.

If the source is non-axisymmetric, it has been confirmed that it needs to be simulated by changing the value of m.
(https://meep.readthedocs.io/en/latest/Python_Tutorials/Cylindrical_Coordinates/#nonaxisymmetric-dipole-sources)
In other words, several simulations need to be defined, but I think I can only give one simulation to the optimization problem.
I wonder how to define the optimization problem.

**Comment by stevengj** [comment]
The objective function that you pass to the optimization algorithm can be a function of as many simulations as you want.
